internal/manifests/collector/collector.go

```diff
@@ -98,6 +98,24 @@ func Build(params manifests.Params) ([]client.Object, error) {
 		return nil, errors.Join(w...)
 	}
 
+	roles, err := Role(params)
+	if err != nil {
+		return nil, err
+	}
+	// NOTE: we cannot just unpack the slice, the type checker doesn't coerce the type correctly.
+	for _, role := range roles {
+		resourceManifests = append(resourceManifests, role)
+	}
+
+	roleBindings, err := RoleBinding(params)
+	if err != nil {
+		return nil, err
+	}
+	// NOTE: we cannot just unpack the slice, the type checker doesn't coerce the type correctly.
+	for _, roleBinding := range roleBindings {
+		resourceManifests = append(resourceManifests, roleBinding)
+	}
```

Comment:
I'd prefer to fold these into the manifest factory framework, but the idea of changing all the current functions to return slices doesn't seem appealing. Maybe we can instead do the conversion in `manifests.Factory`? WDYT @jaronoff97 ?

Reply:
Yeah i think modifying the manifest factory to have a method that allows returning a slice makes sense to me. You could accomplish that in a similar way we do the factory without error today.

Reply:
I will try. Thanks!